model/flow/epoch.go

```diff
@@ -332,3 +335,7 @@ func (es *EpochStatus) Phase() (EpochPhase, error) {
 	}
 	return EpochPhaseCommitted, nil
 }
+
+func (es *EpochStatus) HasPrevious() bool {
```

Comment:
This is the imperative way. A functional way is to return both the previous and hasPrevious, so if there is previous, then you won't read the second value; if there is previous, then here you go, the second value is the "proof" that previous exist:
```
func (es *EpochStatus) HasPrevious() (bool,  <the type of PreviousEpoch>) {
  has := es.PreviousEpoch.SetupID != ZeroID && es.PreviousEpoch.CommitID != ZeroID
  return has, es.PreviousEpoch
}
```

Reply:
What is the benefit to the functional way here? If we can call `HasPrevious` we already have access to `PreviousEpoch`.